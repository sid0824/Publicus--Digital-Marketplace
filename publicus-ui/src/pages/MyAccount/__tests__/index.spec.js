import toJson from 'enzyme-to-json';
import React from "react";
import {shallow, configure} from 'enzyme';
import { message, Input } from 'antd';
import Adapter from 'enzyme-adapter-react-16';
import {MyAccount} from "../index";
import {API, Auth} from 'aws-amplify';
import axios from "axios";

configure({ adapter: new Adapter() });

describe('MyAccount', () => {

  beforeEach(() => {
    axios.put = jest.fn().mockResolvedValue(null);
    API.get = jest.fn().mockResolvedValue(null);
    Auth.changePassword = jest.fn().mockResolvedValue(null);
    Auth.currentAuthenticatedUser = jest.fn().mockResolvedValue(null);
    message.success = jest.fn();
    message.error = jest.fn();
  });

  afterEach(() => {
    axios.put.mockReset();
    API.get.mockReset();
    Auth.changePassword.mockReset();
    Auth.currentAuthenticatedUser.mockReset();
    message.success.mockReset();
    message.error.mockReset();
  });

  test('should render my account page', () => {
    const wrapper = shallow(
      <MyAccount />
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('should update account password', async () => {
    const wrapper = shallow(
      <MyAccount />
    );
    wrapper.find(Input.Password).at(0).props().onChange({
      target: {
        value: 'PASSWORD'
      }
    });
    wrapper.find(Input.Password).at(1).props().onChange({
      target: {
        value: 'PASSWORD'
      }
    });
    await wrapper.instance().changePassword();
    expect(Auth.changePassword.mock.calls[0]).toMatchSnapshot();
    expect(message.success).toHaveBeenCalledWith('Successfully changed password');
  });

  test('should give error msg on bad password update', async () => {
    const wrapper = shallow(
      <MyAccount />
    );
    Auth.changePassword.mockImplementation(() => {
      throw new Error('ERROR')
    });
    await wrapper.instance().changePassword();
    expect(message.error).toHaveBeenCalledWith('Failed to change password');
  });

  test('should handle uploading avatar', async () => {
    const wrapper = shallow(
      <MyAccount />
    );
    wrapper.instance().uploadInput = {};
    wrapper.instance().uploadInput.files = [{
      name: 'fileName.png'
    }];
    API.get.mockResolvedValue({ putUrl: 'PUT_URL' });
    axios.put.mockResolvedValue(null);
    await wrapper.instance().handleUpload();
    expect(message.success).toHaveBeenCalledWith('Successfully uploaded file')
  });

  test('should only allow images to be uploaded', async () => {
    const wrapper = shallow (
      <MyAccount />
    );
    wrapper.instance().uploadInput = {};
    wrapper.instance().uploadInput.files = [{
      name: 'fileName.docx'
    }];
    await wrapper.instance().handleUpload();
    expect(message.error).toHaveBeenCalledWith('Upload a valid image')
  });

});