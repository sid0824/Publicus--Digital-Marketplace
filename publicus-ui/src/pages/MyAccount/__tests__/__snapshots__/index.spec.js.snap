// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccount should render my account page 1`] = `
<div
  style={
    Object {
      "marginLeft": "calc(50% - 300px)",
      "marginRight": "calc(50% - 300px)",
      "marginTop": 10,
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      <h2
        style={
          Object {
            "marginRight": 50,
          }
        }
      >
        User Avatar
      </h2>
      <div
        style={
          Object {
            "height": 200,
            "marginBottom": 12,
            "width": 200,
          }
        }
      >
        <Component
          height={200}
          width={200}
        />
      </div>
      <input
        onChange={[Function]}
        type="file"
      />
    </div>
  </div>
  <br />
  <br />
  <br />
  <h2>
    Change Password
  </h2>
  Current Password
  <br />
  <Password
    action="click"
    inputPrefixCls="ant-input"
    onChange={[Function]}
    placeholder="Enter your current password"
    prefixCls="ant-input-password"
    style={
      Object {
        "width": 600,
      }
    }
    value=""
    visibilityToggle={true}
  />
  <br />
  <br />
  New Password
  <br />
  <Password
    action="click"
    inputPrefixCls="ant-input"
    onChange={[Function]}
    placeholder="Enter your new password"
    prefixCls="ant-input-password"
    style={
      Object {
        "width": 600,
      }
    }
    value=""
    visibilityToggle={true}
  />
  <br />
  <br />
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    type="primary"
  >
    Submit
  </Button>
</div>
`;

exports[`MyAccount should update account password 1`] = `
Array [
  null,
  "PASSWORD",
  "PASSWORD",
]
`;
